### ISSUE TITLE
Restrict `GenericTypeExprOperand`

### ISSUE THREAD

**Issue by brettz9** [jsdoctypeparser/jsdoctypeparser#99]
A user reported an issue at that generic type expressions are quite broad in the operand before the type list, e.g., `Object<Object>` and even `string<Object>` are being accepted.

While `Promise` and of course `Array` make sense (or user-defined types which are promise/array subclasses), as well as Closure generics (which jsdoc inherits, though without `@template` support) which takes a class and indicates a subset whose type doesn't necessarily indicate a direct property as with `Object` (see https://github.com/google/closure-compiler/wiki/Generic-Types ).

But it seems in any case that `GenericTypeExprOperand` is too permissive now as it allows generics based on even string literals.

**Comment by ExE-Boss** [comment]
Yeah, primitives really shouldn’t be allowed by `GenericTypeExprOperand`.
